proxmoxtf/resource/vm.go

```diff
@@ -777,9 +780,18 @@ func VM() *schema.Resource {
 							Type:        schema.TypeString,
 							Description: "The datastore id",
 							Optional:    true,
-							ForceNew:    true,
 							Default:     dvResourceVirtualEnvironmentVMInitializationDatastoreID,
 						},
+						mkResourceVirtualEnvironmentVMInitializationInterface: {
+							Type:             schema.TypeString,
+							Description:      "The IDE interface on which the CloudInit drive will be added",
+							Optional:         true,
+							Default:          dvResourceVirtualEnvironmentVMInitializationInterface,
+							ValidateDiagFunc: validator.IDEInterface(),
+							DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
```

Comment:
Just curious what use case prompted adding this suppression?

Reply:
The suppression prevents the provider from trying to update the resource if it has been initialized with no interface specified. When re-applying, the resource definition will still have an empty `interface` value, but the state (and the Proxmox data) will indicate some `ideX` interface. Without the suppress, it would be detected as changing from `""` to `"ideX"` every time.